### PR TITLE
feat: add Langfuse prompt label support with improved parsing

### LINKED ISSUE #4825: Allow fetching langfuse prompts by label
**Is your feature request related to a problem? Please describe.**
It would be nice to iterate on the prompts without affecting production. To test the latest version of a prompt, one needs to fetch it by the `latest` label instead of a hardcoded version number.

**Describe the solution you'd like**
Langfuse already [supports](https://api.reference.langfuse.com/#tag/prompts/get/api/public/v2/prompts/{promptName}) fetching prompts by label. So this should be an easy add.

### PR DESCRIPTION
**Pull request by mldangelo**
## Summary

This PR adds support for fetching Langfuse prompts by label, addressing #4825.

### ✅ Core Feature Implementation
- Added label-based prompt fetching: `langfuse://prompt-name@label:type`
- Maintains backward compatibility with version-based syntax
- Updated Langfuse integration to pass label parameter to SDK

### 🔧 Improvements Made During Review
1. **Enhanced Parsing Logic**
   - Fixed parsing to correctly handle prompt IDs containing `@` symbols
   - Uses regex to identify the last `@` followed by a label pattern
   - Now supports complex IDs like `email@support@production`

2. **Better Error Handling**
   - Added contextual error messages that specify whether failure was for label or version
   - Makes debugging easier by providing clear error context

3. **Comprehensive Testing**
   - Added 18 test cases covering various scenarios
   - Tests for edge cases including multiple `@` symbols
   - Tests for improved error messages

4. **Documentation**
   - Added examples for label-based syntax
   - Created example configuration in `examples/langfuse-labels/`
   - Updated integration docs with best practices

## Usage Examples

```yaml
prompts:
  # Reference by label (recommended for production)
  - langfuse://my-prompt@production
  - langfuse://chat-prompt@staging:chat
  - langfuse://email-template@latest:text
  
  # Still supports version-based references
  - langfuse://my-prompt:3:text
```

## Benefits
- Deploy new prompt versions without code changes
- Use different prompts for different environments
- A/B test prompt versions
- Quick rollbacks via Langfuse UI

## Testing
All tests pass:
- `test/evaluatorHelpers.langfuse.test.ts` - 18 tests
- `test/integrations/langfuse.test.ts` - 14 tests

Closes #4825

### CODE REVIEW

File: src/evaluatorHelpers.ts [hunk 1]
```diff
@@ -276,17 +276,45 @@ export async function renderPrompt(
...
+      helper = versionMatch[1];
+      version = versionMatch[2];
+      if (versionMatch[3]) {
+        promptType = versionMatch[3] as 'text' | 'chat';
+      }
+    } else {
+      // Simple prompt-id only
+      helper = langfusePrompt;
+    }
+
```
Comment (aandis) [inline src/evaluatorHelpers.ts:321]: nit: an alternative way to achieve this would have been to leverage the fact that labels are always [strings](https://api.reference.langfuse.com/#tag/prompts/get/api/public/v2/prompts/{promptName}) and versions are always [integers](https://api.reference.langfuse.com/#tag/prompts/get/api/public/v2/prompts/{promptName}) and you can only specify one of them. This would then require minimal changes as the current format `langfuse://my-prompt:<integer_version_or_label_string>:(text|chat)` would work with both.

Comment (aandis) [reply src/evaluatorHelpers.ts:321]: But regardless, thank you implementing this so quickly!

**Comment by aandis** [comment]
@mldangelo could you please release this.
